compiler/rustc_middle/src/ty/instance.rs

```diff
@@ -163,6 +163,22 @@ impl<'tcx> InstanceDef<'tcx> {
         }
     }
 
+    /// Returns the `DefId` of instances which might not require codegen locally.
+    pub fn maybe_codegen_locally_def_id(self) -> Option<DefId> {
```

Comment:
This function name confuses me. I get why it is named this way after reading the documentation, the use sites and the body, but maybe we can come up with something less confusing. I came up with the monster `def_id_if_not_guaranteed_local_codegen`.

Reply:
I agree that this name is better, I'll make this change as soon as I'm back on my rustc machine.

Reply:
I've fixed this now.